change `raco pkg new` to generate both MIT and APACHE LICENSE files (replacing LGPL)

spdegabrielle opened the pull request.

sorawee commented on racket/collects/pkg/private/new.rkt:
```diff
@@ -62,8 +62,8 @@
-      ;; LICENSE.txt
-      (with-output-to-file "LICENSE.txt"
+      ;; LICENSE files
+      (with-output-to-file "LICENSE-MIT"
```
The text below is still that of LGPL

spdegabrielle replied:
My mistake. Fixed.

sorawee commented on racket/collects/pkg/private/new.rkt:
```diff
@@ -77,6 +77,215 @@ particular means that you must release the source code for the
+
+   END OF TERMS AND CONDITIONS
+
+   APPENDIX: How to apply the Apache License to your work.
```
Please take the instruction appendix off (until "identification within third-party archives.")

spdegabrielle replied:
done

spdegabrielle commented:
@sorawee Thank you for reviewing so quickly.

sorawee commented on racket/collects/pkg/private/new.rkt:
```diff
@@ -62,20 +62,229 @@
+        (lambda () (expand/display #<<EOS
+MIT License
+
+Copyright (c) <<year>> <<name>> <<user>>
```
I'm not sure what's the best way to deal with this line, but the lack of punctuation between `<<name>>` and `<<user>>` looks... incorrect.

```
Copyright (c) 2019 hello-world sorawee
```

spdegabrielle replied:
fixed - now has name on first line

spdegabrielle commented:
@samth licence file generation change suggested by @sorawee 
all mistakes are mine, thank you to @sorawee for my mistakes. And @greghendershott too.

Added  `(displayln "Generating LICENSE-APACHE and LICENSE-MIT files. You are free to change the license.")`

In line with the guidance at https://www.apache.org/licenses/LICENSE-2.0#apply I've changed the LICENSE-APACHE file generation.

bennn commented on racket/collects/pkg/private/new.rkt:
```diff
@@ -62,20 +62,51 @@
-      ;; LICENSE.txt
-      (with-output-to-file "LICENSE.txt"
+      ;; LICENSE files
+      (displayln "Generating LICENSE-APACHE and LICENSE-MIT files. You are free to change the license.")
```
Is there any reason these don't use the `.txt` file extension?

spdegabrielle replied:
Hi @benn

@sorawee asked @samth and Sam said no to having a txt extension.
 
See https://github.com/racket/racket/issues/1570#issuecomment-501449180

Kind regards, 
Stephen

spdegabrielle commented:
Updated the generated main.rkt file to refer to the new license files included.

NlightNFotis commented on racket/collects/pkg/private/new.rkt:
```diff
@@ -222,9 +222,8 @@ EOS
 ;;   $ raco docs <<name>>
 ;;
-;; For your convenience, we have included a LICENSE.txt file, which links to
-;; the GNU Lesser General Public License.
```
Is this bit of information user facing somewhere? Perhaps `raco` output? or is it only a source code comment?

NlightNFotis replied:
Ignore this comment, I just noticed in an earlier commit that this is user facing. It's weird though that the comment is down below. Is there any reason for it being located where it is, or is it misplaced?

spdegabrielle replied:
for other reviewers - this section was also changed to refer to MIT and APACHE licenses,

``` 
;; To view documentation:
;;   $ raco docs <<name>>
;;
;; For your convenience, we have included LICENSE-MIT and LICENSE-APACHE files.
;; If you would prefer to use a different license, replace those files with the
;; desired license.
```

NlightNFotis commented:
Also, before this gets merged in, can the multiple `Update new.rkt` commits be squashed into one with a more descriptive commit message?

spdegabrielle commented:
squashed into new PR #2701
